opm/simulators/wells/WellAssemble.cpp

```diff
@@ -179,9 +179,13 @@ assembleControlEqProd(const WellState& well_state,
                 active_rates[pu.phase_pos[canonical_phase]] = rates[canonical_phase];
             }
         }
-        auto rCoeff = [this](const RegionId id, const int region, std::vector<double>& coeff)
+        auto rCoeff = [this, group_state](const int id, const int region, const std::optional<std::string>& prod_gname, std::vector<double>& coeff)
```

Comment:
Just to be clear: This will create a copy of the `group_state` inside the lambda.  Is that what you want?  If not, the capture should be `&group_state` instead.  Also, is there a reason for switching from `RegionId` to `int` here?  That potentially introduces needless type conversions.

Reply:
Thanks, I will update the code. @tskille has tested the code and confirms that is indeed fixes the unexpected results for RESV controlled wells for under saturated oil. i.e. this should be merged when the code is fixed.